exonum-java-binding/common/src/main/java/com/exonum/binding/common/serialization/json/ZonedDateTimeJsonSerializer.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.exonum.binding.common.serialization.json;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import java.lang.reflect.Type;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * {@link ZonedDateTime} string serializer. Used to serialize/deserialize ZonedDateTime values
+ * from/to strings in 'yyyy-MM-dd HH:mm:ss Z' format.
+ *
+ * <p>All method arguments are non-null by default.
+ */
+final class ZonedDateTimeJsonSerializer implements JsonSerializer<ZonedDateTime>,
+    JsonDeserializer<ZonedDateTime> {
+
+  private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter
+      .ofPattern("yyyy-MM-dd HH:mm:ss Z");
+
+  /**
+   * Serialize ZonedDateTime to JsonElement.
```

Comment:
I'd remove Javadocs for overrides as they are not required (and these don't add much).

Reply:
I'd leave those be, especially since there is new information compared with overridden methods (`@throws ...`). Plus for consistency - we have javadocs in other serializers (for HashCode and PublicKey, for example. Not others though).

Reply:
We can inherit docs:
```
  /**
   * @inheritDoc
   * @throws NullPointerException in case of src is null
   */
```
Also, next one works well:
```
  /**
   * @throws NullPointerException in case of src is null
   */
```

Reply:
We have a convention of non-null by default, so, again, I'd not specify it here explicitly (and complicate with inheritDoc). 

However, yes, they can remain.

Reply:
We have `<p>All method arguments are non-null by default.` in class' Javadoc, so removed these.